src/component/page/bruker-detaljer/tildel-veileder-modal/TildelVeilederModal.tsx

```diff
@@ -0,0 +1,144 @@
+import { Alert, Button, Detail } from '@navikt/ds-react'
+import React, { useEffect, useMemo, useState } from 'react'
+import { TilgjengeligVeileder, Veileder } from '../../../../api/data/veileder'
+import { MultiValue, SingleValue } from 'react-select'
+import { lagBrukerNavn } from '../../../../utils/bruker-utils'
+
+import styles from './TildelVeilederModal.module.scss'
+import { isPending, isResolved, usePromise } from '../../../../utils/use-promise'
+import { AxiosResponse } from 'axios'
+import { hentTilgjengeligeVeiledere, tildelVeilederForDeltaker } from '../../../../api/tiltak-api'
+import { TiltakDeltakerDetaljer } from '../../../../api/data/deltaker'
+import { SelectField, SelectOption } from '../../../felles/select-field/SelectField'
+import { BaseModal } from '../../../felles/base-modal/BaseModal'
+
+
+interface Props {
+	deltaker: TiltakDeltakerDetaljer
+	veileder: Veileder | undefined
+	medveiledere: Veileder[]
+	open: boolean
+	onClose: () => void
+	onSubmit: (veiledere: Veileder[]) => void
+}
+
+const maksMedveiledere = 3
+
+export const TildelVeilederModal = (props: Props): React.ReactElement => {
+	const [ tilgjengeligeVeiledere, setTilgjengeligeVeiledere ] = useState<TilgjengeligVeileder[]>([])
+	const [ veilederValg, setVeilederValg ] = useState<SelectOption[]>()
+	const [ veileder, setVeileder ] = useState<TilgjengeligVeileder | undefined>()
+	const [ medveiledere, setMedveiledere ] = useState<TilgjengeligVeileder[]>([])
+	const [ veilederError, setVeilederError ] = useState(false)
+
+	const tildelVeilederePromise = usePromise<void>()
+	const tilgjengeligeVeilederePromise = usePromise<AxiosResponse<TilgjengeligVeileder[]>>(
+		() => hentTilgjengeligeVeiledere(props.deltaker.gjennomforing.id)
+	)
+
+
+	useMemo(() => {
+		if (isResolved(tilgjengeligeVeilederePromise)) {
+			setTilgjengeligeVeiledere(tilgjengeligeVeilederePromise.result.data)
+		}
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ tilgjengeligeVeilederePromise.result ])
+
+	useEffect(() => {
+		if (props.veileder) {
+			setVeileder(props.veileder)
+		}
+		if (props.medveiledere) {
+			setMedveiledere(props.medveiledere)
+		}
+	}, [ props.veileder, props.medveiledere ])
+
+	useEffect(() => {
+		const ansattIder = medveiledere.map(v => v.ansattId)
```

Comment:
Vet ikke om jeg skjønte denne useEffecten helt. Er dette for å filtrere bort fra lista de som allerede er valgt?

Reply:
Det er riktig.